mutiny-core/src/nodemanager.rs

```diff
@@ -736,18 +795,57 @@ impl NodeManager {
         Ok(details_opt.map(|(d, _)| d))
     }
 
+    fn add_onchain_labels(
+        &self,
+        address_labels: HashMap<Address, Vec<String>>,
+        tx: bdk::TransactionDetails,
+    ) -> TransactionDetails {
+        // find the first output address that has a label
+        let labels = tx
+            .transaction
+            .clone()
+            .unwrap() // safe because we call with list_transactions(true)
```

Comment:
Ugh, well now this isn't always going to be the case... A comment on the function itself might be necessary, or safely handle this and don't add any labels.